docs/data-structures/depth-of-tree.mdx

```diff
@@ -0,0 +1,262 @@
+                    //else we change the current node pointer with pointer of left child node pointer
+                    else
+                     temp = temp->right_child;
+                }
+            }
+        }
+    }
+
+    int depth(node *level)
+    {
+        if(level == NULL)
+        {
+            return 0;
+        }
+        //depth of left subtree
+        int left_depth = depth(level->left_child);
+        //depth of right subtree
+        int rigth_depth = depth(level->right_child);
+        // depth of tree will max of depth of left subtree and depth of right subtree adding current node;
+        return  max(left_depth,rigth_depth)+1;
+    }
+	
+    //depth of whole tree
+    int depth_of_tree()
+    {
+        return(depth(root));
+    }
+};
+int main() {
+    bin_tree t;
+    int n;
+    cout<<"Enter number of elements in tree: ";
+    cin>>n;
+    cout<< "Enter elements to insert in  tree: ";
+    while(n--)
+    {	
+        int k; cin>>k;
+        t.insert_data(k);
+    }
+    cout<<"\nDepth of tree:"<<t.depth_of_tree()<<endl;
+    return 0;	
+}
+```
+
+</TabItem>
+
+<TabItem value="py">
+
+```python
+class TreeNode:
+    def __init__(self, val):
+        self.val = val
+        self.left = None
+        self.right = None
+
+
+class Tree:
+    def __init__(self, root):
+        self.root = TreeNode(root)
+
+    def get_node_depth(self, node, val, depth):
+        if node is None:
+            return -1
+
+        if node.val is val:
+            return depth
+
+        l_depth = self.get_node_depth(node.left, val, depth+1)
+        r_depth = self.get_node_depth(node.right, val, depth+1)
+
+        return max(l_depth, r_depth)
+
+    def get_maximum_depth(self, node, depth):
+        if node is None:
+            return -1
+
+        l_depth = 1 + self.get_maximum_depth(node.left, depth+1)
+        r_depth = 1 + self.get_maximum_depth(node.right, depth+1)
+
+        return max(l_depth, r_depth)
+
+tree = Tree(1)
+
+tree.root.left = TreeNode(2)
+tree.root.right = TreeNode(3)
+tree.root.left.left = TreeNode(4)
+tree.root.left.right = TreeNode(5)
+tree.root.right.left = TreeNode(6)
+tree.root.right.right = TreeNode(7)
+
+# Printing the depth of each node
+for i in range(1, 8):
+    print(tree.get_node_depth(tree.root, i, 0))
+   
+# Printing the maximum depth of the tree
+print("Maximum Depth: ", tree.get_maximum_depth(tree.root, 0))
+
+# Print -1 for the depth if the node is not present in the tree
+print("If element not present", tree.get_node_depth(tree.root, 10, 0))
+```
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Sample Input
+```
+Enter number of elements in tree: 7
+Enter elements to insert in  tree: 50 60 40 55 30 70 45 
+```
+
+#### Sample Output
+```
+Depth of tree : 3
+
+         50                         level 1
+      /     \
+     /       \
+    40        60                    level 2
+   /   \     /   \
+  /     \   /     \
+30    45  55      70                level 3
```

Comment:
Is this even a part of the output?

Reply:
No, it's not. I just added it for the explanation, but I'll remove it now